redis-om-spring/src/main/java/com/redis/om/spring/indexing/RediSearchIndexer.java

```diff
@@ -815,4 +783,43 @@ private String getFieldPrefix(String prefix, boolean isDocument) {
     String chain = (prefix == null || prefix.isBlank()) ? "" : prefix + ".";
     return isDocument ? "$." + chain : chain;
   }
+
+  private void registerAlias(Class<?> cl, String fieldName, String alias) {
+    entityClassFieldToAlias.put(Tuples.of(cl, fieldName), alias);
+  }
+
+  private FieldName buildFieldName( //
+    java.lang.reflect.Field field, String prefix, boolean isDocument, Optional<String> maybeAlias,
+    Optional<Integer> maybeArrayIndex) {
+    SerializedName serializedName = field.getAnnotation(SerializedName.class);
+    Indexed indexed = field.getAnnotation(Indexed.class);
+    String fname = (serializedName != null) ? serializedName.value() : field.getName();
+
+    TypeInformation<?> typeInfo = TypeInformation.of(field.getType());
+    String fieldPrefix = getFieldPrefix(prefix, isDocument);
+
+    String index = maybeArrayIndex.isPresent() && (maybeArrayIndex.get() != Integer.MIN_VALUE) ?
+      ".[" + maybeArrayIndex.get() + "]" :
+      "[*]";
+
+    boolean needsPostfix = (isDocument && typeInfo.isCollectionLike() && !field.isAnnotationPresent(
+      JsonAdapter.class) && (indexed != null && !indexed.schemaFieldType().equals(SchemaFieldType.VECTOR)));
+    String fieldPostfix = needsPostfix ? index : "";
+
+    String name = fieldPrefix + fname + fieldPostfix;
+
+    String alias = maybeAlias.isEmpty() || maybeAlias.get().isBlank() ?
+      QueryUtils.searchIndexFieldAliasFor(field, prefix) :
+      maybeAlias.get();
+
+    FieldName fieldName = FieldName.of(name);
+    fieldName = fieldName.as(alias);
+
+    return fieldName;
```

Comment:
```suggestion
    return FieldName.of(name).as(alias);
```
You know you can do this, right? :smile: 
Or do you prefer the other one?